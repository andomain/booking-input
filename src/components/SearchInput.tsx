import React, { FC, ChangeEvent } from 'react'

interface SearchInputProps {
  id: string;
  name: string;
  label: string;
  placeholder: string;
  value: string;
  onChange: (val: string) => void;
}

const SearchInput: FC<SearchInputProps> = ({ id, name, label, onChange, placeholder, value }) => {

  const changeHandler = (e: ChangeEvent<HTMLInputElement>): void => {
    const { target: { value } } = e;
    onChange(value);
  }

  return (
    <div className="SearchInput">
      <label htmlFor={id}>{label}</label>
      <input id={id} name={name} onChange={changeHandler} type="text" placeholder={placeholder} value={value} />
    </div>
  )
}

export default SearchInput
