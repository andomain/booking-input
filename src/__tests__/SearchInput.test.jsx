import React from "react";
import { shallow } from "enzyme";
import SearchInput from "../components/SearchInput";

const TEST = {
  ID: "test-id",
  NAME: "test-name",
  LABEL: "test-label",
  PLACEHOLD: "test-placehold",
  VALUE: "test-value"
};

describe("<SearchInput />", () => {
  let wrapper;
  let input;
  let label;

  const changeHandlerMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <SearchInput
        id={TEST.ID}
        name={TEST.NAME}
        label={TEST.LABEL}
        onChange={changeHandlerMock}
        placeholder={TEST.PLACEHOLD}
        value={TEST.VALUE}
      />
    );
    input = wrapper.find("input");
    label = wrapper.find("label");
  });

  it("contains a text <input />", () => {
    expect(input.length).toEqual(1);
    expect(input.props().type).toEqual("text");
  });

  it("should have an id & corresponding label for screenreaders", () => {
    expect(input.props().id).toEqual(TEST.ID);
    expect(label.props().htmlFor).toEqual(TEST.ID);
  });

  it('should display a label', () => {
    expect(label.text()).toEqual(TEST.LABEL);
  })

  it("should display a placeholder", () => {
    expect(input.props().placeholder).toEqual(TEST.PLACEHOLD);
  });

  it("should display the value prop", () => {
    expect(input.props().value).toEqual(TEST.VALUE);
  });

  // TODO: Test focus state
  // This is not essential as the required behaviour is browser default
  // it('should focus on the input element when clicked', async () => {
  //   expect(document.activeElement).toEqual(input);
  // });

  it("should fire onChange prop when value changes", () => {
    const event = {
      preventDefault() {},
      target: {
        value: "testValue"
      }
    };

    input.simulate("change", event);
    expect(changeHandlerMock).toHaveBeenCalledWith("testValue");
  });
});
