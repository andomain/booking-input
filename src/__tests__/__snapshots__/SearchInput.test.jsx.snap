// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should render correctly 1`] = `ShallowWrapper {}`;
